compose/bin/start

```diff
@@ -1,4 +1,20 @@
 #!/bin/bash
+if ruby -ryaml -e "" 2>/dev/null; then
+  VOLUME_LIST=`ruby -ryaml -e 'YAML.load(ARGF)["services"]["app"]["volumes"].each{|x| puts x.split(":")[0]}' docker-compose.dev.yml`
```

Comment:
I absolutely love what you are doing here. Is there any chance to do this in pure bash though rather than ruby? I *think* everyone has ruby on their systems (Windows WSL, Ubuntu, etc?), but I wasn't sure.

Reply:
I just found the most concise solution for mac. Let me show the pure bash (`sed` and `awk`) way:
```
function getVolumes {
  local s='[[:space:]]*' w='[a-zA-Z0-9_]*' fs=$(echo @|tr @ '\034')
  sed -ne "s|,$s\]$s\$|]|" \
      -e ":1;s|^\($s\)\($w\)$s:$s\[$s\(.*\)$s,$s\(.*\)$s\]|\1\2: [\3]\n\1  - \4|;t1" \
      -e "s|^\($s\)\($w\)$s:$s\[$s\(.*\)$s\]|\1\2:\n\1  - \3|;p" $1 | \
  sed -ne "s|,$s}$s\$|}|" \
      -e ":1;s|^\($s\)-$s{$s\(.*\)$s,$s\($w\)$s:$s\(.*\)$s}|\1- {\2}\n\1  \3: \4|;t1" \
      -e    "s|^\($s\)-$s{$s\(.*\)$s}|\1-\n\1  \2|;p" | \
  sed -ne "s|^\($s\):|\1|" \
      -e "s|^\($s\)-$s[\"']\(.*\)[\"']$s\$|\1$fs$fs\2|p" \
      -e "s|^\($s\)-$s\(.*\)$s\$|\1$fs$fs\2|p" \
      -e "s|^\($s\)\($w\)$s:$s[\"']\(.*\)[\"']$s\$|\1$fs\2$fs\3|p" \
      -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p" | \
  awk -F$fs '{
    indent = length($1)/2;
    vname[indent] = $2;
    for (i in vname) {if (i > indent) {delete vname[i]; idx[i]=0}}
    if (length($2) == 0) {vname[indent] = ++idx[indent] };
    if (length($3) > 0) {
      vn=""; for (i=0; i<indent; i++) {vn = (vn)(vname[i])("_")}
      if ("services_app_volumes_" == vn) {
         print substr($3 ,0 , match($3,":")-1)
      }
    }
  }'
}

getVolumes docker-compose.dev.yml
```
Ref: [How can I parse a YAML file from a Linux shell script?](https://stackoverflow.com/questions/5014632/how-can-i-parse-a-yaml-file-from-a-linux-shell-script)
How do you think? Do I use this to replace the ruby ?

Reply:
It remove ruby dependency